src/common.rs

```diff
@@ -104,7 +108,47 @@ pub fn get_data_dirs(bin_path: impl AsRef<Path>) -> Result<(PathBuf, PathBuf), C
         .ok_or(ConfigError::MissingKey("write-data".to_string()))?;
 
     Ok((
-        resolve_path(&read_path, &bin_path)?,
-        resolve_path(&write_path, bin_path)?,
+        resolve_path(&read_path, &bin_path)?.canonicalize()?,
+        resolve_path(&write_path, bin_path)?.canonicalize()?,
     ))
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum ExtractError {
```

Comment:
I believe this error struct should be moved into the `rivets-rs` repo.
It will be re-used in librivets when extracting all the mods, and we should avoid code duplication.

Reply:
yeah eventually some of these helper functions should be moved to rivets-rs, for now I'd just keep them here tho